### DISCUSSION TITLE
Task (Taskfile) for command runner

### DISCUSSION THREAD

**Discussion by pjbull** [drivendataorg/cookiecutter-data-science#378] [Requested tools]
Task is a tool for tracking and managing project-specific commands.

Docs: https://taskfile.dev/
Github: https://github.com/go-task/task

Pros:
 - Binaries and packaged versions for all platforms
 - Straightforward syntax with more specialized options (for loops, branches per os, dependencies)
 - Actively developed

Cons: 
 - `yaml` can be verbose and hard to read
 - does not come pre-installed on most systems
 - will be unfamiliar to many

**Comment by jayqi** [comment]
It looks like someone made a PyPI package for distributing the Task binary. https://pypi.org/project/go-task-bin/

This means we could have it be a dependency for CCDS-generated projects. Though one would still need to install it first before being able to use it to manage project tasks.
